schemas/tests.yaml

```diff
@@ -0,0 +1,233 @@
+---
+
+#
+# JSON Schema definition for tmt `Tests`
+#
+# https://tmt.readthedocs.io/en/stable/spec/tests.html
+#
+
+definitions:
+
+  # https://fmf.readthedocs.io/en/stable/concept.html#identifiers
+  _fmf_id:
+    type: object
+    minProperties: 1
+    properties:
+      url:
+        type: string
+        # TODO: maybe also "format: uri"
+        # together with "pattern" to enforce http instead
+        # of all possible protocols?
+        pattern: "^https?://"
+
+      ref:
+        type: string
+
+      path:
+        type: string
+
+      name:
+        type: string
+        pattern: "^/"
+
+  fmf_id:
+    allOf:
+      - $ref: "tests.yaml#/definitions/_fmf_id"
+
+    additionalProperties: false
+
+    properties:
+      url: true
+      ref: true
+      path: true
+      name: true
+
+  beakerlib_library:
+    allOf:
+      - $ref: "tests.yaml#/definitions/_fmf_id"
+
+    additionalProperties: false
+
+    properties:
+      url: true
+      ref: true
+      path: true
+      name: true
+      nick:
+        type: string
+
+  link:
```

Comment:
I'd probably recommend a small name swap. This definition is called `link` while `link` key schema refers to definition called `link_and_note`. I'd argue that `link` is supposed to support `note` key "out of the box", without any extensions, therefore this definition should have a different name, e.g. `_link`, making space for `link_with_note` to become `link`, more aligned with what it describes.

`_link` is a subschema, sevres only to simplify the actual `link` schema, and without a `note` key support it's not a link as defined in specification.

Reply:
ack